subsys/nsettings/settings_fcb.c

```diff
@@ -0,0 +1,550 @@
+/* Copyright (c) 2019 Laczen
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2015 Runtime Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <string.h>
+
+#include "settings_fcb.h"
+#include "settings_priv.h"
+
+struct settings_fcb_read_fn_arg {
+	struct fcb_entry_ctx *entry_ctx;
+	off_t off;
+};
+
+static int settings_fcb_load(struct settings_store *cs);
+static int settings_fcb_save(struct settings_store *cs, const char *name,
+			     const char *value, size_t val_len);
+
+static struct settings_store_itf settings_fcb_itf = {
+	.csi_load = settings_fcb_load,
+	.csi_save = settings_fcb_save,
+};
+
+int settings_fcb_dst(struct settings_fcb *cf)
+{
+	cf->cf_store.cs_itf = &settings_fcb_itf;
+	settings_dst_register(&cf->cf_store);
+
+	return 0;
+}
+
+int settings_fcb_src(struct settings_fcb *cf)
+{
```

Comment:
a fcb instance initialization moved somewhere.

Reply:
Yes, why is it like this: settings_fcb_src registers a source, if you would like to use multiple fcb areas as sources you first create the multiple fcb areas (similar to what is done in settings_init.c), then you register each of these fcb areas as a source with settings_fcb_src. By the way, maybe it would be better to rename settings_init.c to settings_default.c